Week2/Pascal's_triangle_4_fixes/src/Triangle.java

```diff
@@ -16,32 +17,53 @@ private static Long getFactorial(Integer k) {
 		return factorial;
 	}
 	
-	private static Long getMuntiplication(Integer a, Integer b) {
+	private static Long getMultiplicationAtoB(Integer a, Integer b) {
 		Long result = 1L;
 		for (int i = a; i <= b; i++) {
 			result = i * result;
 		}
 		return result;
 	}
 	
-	private static Long C(int n, int k) {
-		Integer max = Integer.max(k, n-k);
-		Integer min = Integer.min(k, n-k);
-		return getMuntiplication(max + 1, n) / (getFactorial(min));
-	}
-	
-	public void printTriangle(int n) {
+		
+	public void fix1_SoutToSBuilder(int _n) {
 		StringBuilder sb = new StringBuilder();
 		
-		for (int i = 0; i <= n; ++i) {
-			for (int j = 0; j <= i; ++j) {
-				sb.append(C(i,j) + " ");
+		for (int n = 0; n <= _n; ++n) {
+			for (int k = 0; k <= n; ++k) {
+				sb.append(C(n,k) + " ");
 			}
 			sb.append("\n");
 		}
 		System.out.println(sb);
 	}
 	
+	private static Long C(int n, int k) {
+		return getFactorial(n) / (getFactorial(k) * getFactorial(n - k));
+	}
+	
+	public void fix2_FactorialToPartMult(int _n) {
+		StringBuilder triangle = new StringBuilder();
+		
+		for (int n = 0; n <= _n; n++) {
+			for (int k = 0; k <= n; k++) {
+				triangle.append(C_update(n, k) + " ");
```

Comment:
Bad chunk of code - "a" + "b" creates additional StringBuilder inside, consider revising into x.append(y).append(z) instead of x.append(y+z)